### PR TITLE
Reduce image size and add last git version

### PR DESCRIPTION
**Pull request by xavierpuyo**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-
-RUN npm install sfdx-cli prettier prettier-plugin-apex --global
-RUN sfdx --version
-RUN sfdx plugins --core
-
-# Work-around to make java installation work
-# as per https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199
-RUN mkdir -p /usr/share/man/man1
-RUN apt-get install openjdk-8-jdk -y
+RUN apt update && apt install -y git openssh-server ca-certificates openssl jq gettext xmlstarlet build-essential libcurl4-gnutls-dev libexpat1-dev libz-dev libssl-dev make && \
```
Comment (airtonzanon) [inline Dockerfile:3]: Nice thing to minimize to a layer <3

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+    git clone git://git.kernel.org/pub/scm/git/git.git && \
```
Comment (airtonzanon) [inline Dockerfile:4]: Why do we need to update git version for a docker image?

File: Dockerfile [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+    cd .. && \
+    rm -rf git && \
+    npm install sfdx-cli prettier prettier-plugin-apex --global && \
+    sfdx --version && \
+    sfdx plugins --core && \
+    mkdir -p /usr/share/man/man1 && \
+    apt-get install openjdk-8-jre-headless -y && \
+    apt-get purge libcurl4-gnutls-dev libexpat1-dev libz-dev libssl-dev make build-essential -y && \
+    apt-get autoremove -y && \
+    apt-get clean && \
```
Comment (airtonzanon) [inline Dockerfile:17]: <3

File: Dockerfile [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+    git clone git://git.kernel.org/pub/scm/git/git.git && \
+    cd git && \
+    make prefix=/usr/ all && \
+    make prefix=/usr/ install && \
+    cd .. && \
+    rm -rf git && \
+    npm install sfdx-cli prettier prettier-plugin-apex --global && \
+    sfdx --version && \
+    sfdx plugins --core && \
+    mkdir -p /usr/share/man/man1 && \
```
Comment (airtonzanon) [inline Dockerfile:13]: Maybe we should keep the comment 

```
# Work-around to make java installation work
# as per https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199
```

**Review by airtonzanon — CHANGES_REQUESTED** [review]

**Comment by malukenho** [comment]
Can we have more information about how image size is reduced with that patch on the commit message? — which is currently empty.

File: Dockerfile (hunk 2 above)
Comment (xavierpuyo) [reply Dockerfile:4]: the version if git is 2.11 and i would like to use the option -o
https://git-scm.com/docs/git-push/2.11.1
https://git-scm.com/docs/git-push/

as you can see the version 2.11 wasn't included.

Comment (EHER) [reply Dockerfile:4]: You shouldn't depend on the git version from this image.
As far as I know this git included here is just a dependency from npm and may not be available on the future.

If you wanna run something that have as the main command git then you should have your own image for that. So you can release versions for that new image.

In GitLab CI/CD Pipeline it's possible to select a different image per job: https://docs.gitlab.com/ee/ci/yaml/#image

**Comment by EHER** [comment]
I created a pull request changing the base image to alpine: #7 

It may be an alternative for image size. 

It's now with 395MB and will be hard to make it smaller since about 300MB is the sfdx package.
Alpine is about 80MB.

**Comment by EHER** [comment]
We should have another image for prettier with apex plugin.

I'll try to make some time this weekend to create it.

**Comment by EHER** [comment]
This may replace the prettier apex plugin: https://cloud.docker.com/u/codelicia/repository/docker/codelicia/apex-prettier

**Comment by xavierpuyo** [comment]
@EHER I changed my source code with your proposal, and it's working well.
Thanks.
The MR can be deleted

**Pull request closed without merging**